## Dr. Dirac's Statistics Midterm
Grading a multiple choice exam is easy. But how much do multiple choice exams tell us about what a student really knows? Dr. Dirac is administering a statistics midterm exam and wants to use Bayes’ Theorem to help him understand the following:

Given that a student answered a question correctly, what is the probability that she really knows the material?

Dr. Dirac knows the following probabilities based on many years of teaching:
- There is a question on the exam that 60% of students know the correct answer to.
- Given that a student knows the correct answer, there is still a 15% chance that the student picked the wrong answer.
- Given that a student does not know the answer, there is still a 20% chance that the student picks the correct answer by guessing.

Using these probabilities, we can answer the question.

<b>1.In order to use Bayes Theorem, we need to phrase our question as P(A|B).</b>

<b>What is A and B in this case?</b>

In [6]:
# P(A|B) = P(knows the matterial | answer correctly)

<b>2. What is the probability that the student knows the material?</b>

In [7]:
# P(A) = P(knows the material) = 60%
p_knows_the_material = 0.6

<b>3. Given that the student knows the material, what is the probability that she answers correctly?</b>

In [8]:
# P(B|A) = P(answer correctly | knows the material) = 1-0.15
p_answer_correctly_knows_the_material = 1-0.15
p_answer_correctly_knows_the_material

0.85

<b>4. What is the probability of any student answering correctly?</b>

In [9]:
p_not_know_the_material = 0.4
p_answer_correctly_not_know_the_material = 0.2

In [10]:
# P(B) = P(answer correctly)
p_answer_correctly = (p_answer_correctly_knows_the_material * p_knows_the_material) +\
                    (p_answer_correctly_not_know_the_material * p_not_know_the_material)
p_answer_correctly

0.5900000000000001

<b>5. Using the three probabilities and Bayes’ Theorem, calculate P(knows material | answers correctly).</b>

In [11]:
# P(A|B) = P(B|A) * P(A) / P(B)
p_knows_the_material_answer_correctly = (p_answer_correctly_knows_the_material * p_knows_the_material) / (p_answer_correctly)
p_knows_the_material_answer_correctly

0.8644067796610169

The probability she knows the answer 86%, given the correct answer.